pubsub/gochannel/pubsub.go

```diff
@@ -342,12 +346,13 @@ func (s *subscriber) Close() {
 }
 
 func (s *subscriber) sendMessageToSubscriber(msg *message.Message, logFields watermill.LogFields) {
-	s.sending.Lock()
-	defer s.sending.Unlock()
+	s.sendingChannel <- struct{}{}
+	defer func() {
+		<-s.sendingChannel
```

Comment:
`sendMessageToSubscriber` is also called to deliver persisted messages to newly joining subscribers. I'm not sure if this will behave as expected in this case. Can we have some tests to cover it?

Reply:
Sure, I'll check again and add some tests for it.